addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+interface MapType {
+  [key: string]: MapType | string[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createtsSourceFile(fileName: string): ts.SourceFile {
+  return ts.createSourceFile (
+  fileName,
+  fs.readFileSync(fileName).toString(),
+  ts.ScriptTarget.ES6,
+  true
+    );
+}
+
+/**
+ *  Store all the files' path under the given path in a passed in array.
+ *  All the files in the subdirectories of the given path are also included.
+ * 
+ * @param filePath path from where all the files' path are retrieved
+ * @param allFiles array where the files are stored and then returned.
+ * 
+ */
+
+export function readAllDir(filePath: string, allFiles: string[]): any {
+  let files = fs.readdirSync(filePath);
+  files.forEach(file => {
+    if (path.extname(file) === '') {
+      readAllDir(path.join(filePath, file), allFiles);
+    } else {
+      allFiles.push(path.join(filePath, file));
+    }
+  });
+  return allFiles;
+}
+
+/**
+ * Traverse through ts.Node recursively and stores moduleSpecifiers in an array
+ * 
+ * @param {ts.Node} node: Typescript Node of whose AST is being traversed
+ * @param {string[]} moduleSpecifiers: array to store moduleSpecifiers if found.
+ * 
+ */
+export function getModuleSpecifier(node: ts.Node, moduleSpecifiers: any[]): any {
```

Comment:
If possible, don't use `any`.  Your `@param` description seems to indicate this is